crates/core/src/dht/dht_store.rs

```diff
@@ -317,15 +317,19 @@ impl DhtStore {
 
 impl GetContent for DhtStore {
     fn get_raw(&self, address: &Address) -> HcResult<Option<Content>> {
-        Ok((*self.content_storage.read().unwrap()).fetch(address)?)
+        if self.holding_list.contains(address) {
+            Ok((*self.content_storage.read().unwrap()).fetch(address)?)
+        } else {
+            Ok(None) // if its not in the holding list it is as if we don't have it
+        }
     }
 }
 
 impl AddContent for DhtStore {
     fn add<T: AddressableContent>(&mut self, content: &T) -> HcResult<()> {
-        (*self.content_storage.write().unwrap())
-            .add(content)
-            .map_err(|e| e.into())
+        (*self.content_storage.write().unwrap()).add(content)?;
+        self.mark_entry_as_held(content);
```

Comment:
Ok further investigations show the error is caused by this line. The assumption is that everything added to the DHT belongs on the hold list but there must be some exceptions to this (likely a Dna | CapTokenGrant | CapTokenClaim )

I don't think this code is at fault but rather it has revealed a fault in another part of the code base where something was added to the CAS via the DHT store that was not meant to be shared